airflow/providers/amazon/aws/hooks/sagemaker.py

```diff
@@ -647,28 +649,28 @@ def describe_endpoint(self, name: str) -> dict:
 
     def check_status(
         self,
-        job_name: str,
+        resource_name: str,
```

Comment:
I just wondering is this changes could be classified as breaking changes or not?
It is a small chance that user might use this public method in their code and define as arguments as keywords

```python
SageMakerHook.check_status(
    job_name="foo-bar",
    key="spam",
    describe_function=some_callable
    check_interval=42
)
```
And after this changes their got:
`TypeError: got an unexpected keyword argument 'job_name'`

Reply:
I'm really waiting for Python 3.7 EOL date with positional only arguments

Reply:
ah that's a good point... I can keep it as job_name to avoid that, there is no strong need to rename it.
I can also add a small comment about the fact that it can be used to check more than jobs.

tbh I think this method should have been private, it's mostly a helper, only used in `wait_for_completion` cases, but now it's too late to change that.